File: ca2_midterm/test2.js
function f( A, B, C, n, r) {
  for(let k=0; k<r; k++) {
    for(let i=0; i<n; i++) {
      for(let j=0; j<n; j++) {
        C[j*n+i] = A[j*n+i]+B[j*n+i];
      }
    }
  }
}

function main() {
  let n = 128;
  let r = 0xffff;
  let A = [...Array(n * n).keys()].map((_) => { return 0; });
  let B = [...Array(n * n).keys()].map((_) => { return 0; });
  let C = [...Array(n * n).keys()].map((_) => { return 0; });
  f(A, B, C, n, r);
}

main();

File: ca2_midterm/Test2.java
class Test2 {
public static void f(long[] A, long[] B, long[] C, int n, long r) {
  for(long k=0; k<r; k++) {
    for(int i=0; i<n; i++) {
      for(int j=0; j<n; j++) {
        C[j*n+i] = A[j*n+i]+B[j*n+i];
      }
    }
  }
}
public static void main(String arg[]) {
	int n = 128;
	long r = 0xffff;
	long[] A = new long[n*n];
	long[] B = new long[n*n];
	long[] C = new long[n*n];
	f(A, B, C, n, r);
}
}
